File: MiniDOMWebLibs1/MiniDOM/WebSiteDebugInfo.vb
Imports Microsoft.VisualBasic
Imports minidom
Imports minidom.Sistema
Imports minidom.Databases
Imports minidom.Anagrafica
Imports System.Web

Partial Class WebSite

    ''' <summary>
    ''' Questa classe serve per stabilire un sistema di comunicazione ad eventi 
    ''' con 
    ''' </summary>
    ''' <remarks></remarks>
    Public Class WebSiteDebugInfo
        Inherits SysDebuInfo

        Public Sub New()
            ' DMDObject.IncreaseCounter(Me)
        End Sub

        Public Overrides Sub Initialize()
            MyBase.Initialize()

            Me.Notes &= ""
            Me.Notes &= minidom.Sistema.Strings.NChars(80, "-") & vbNewLine
            Me.Notes &= "SESSIONI ATTIVE" & vbNewLine
            Me.Notes &= minidom.Sistema.Strings.NChars(80, "-") & vbNewLine

            Dim col As CCollection(Of CSessionInfo) = DirectCast(Sistema.ApplicationContext, WebApplicationContext).GetAllSessions
            For Each info As CSessionInfo In col
                With info
                    Me.Notes &= vbNewLine
                    Me.Notes &= minidom.Sistema.Strings.NChars(80, "-") & vbNewLine

                    Me.Notes &= "SessionID: " & .CurrentSession.SessionID & vbNewLine
                    Me.Notes &= "StartTime: " & Formats.FormatUserDateTime(.CurrentSession.StartTime)
                    Me.Notes &= vbNewLine
                    If (.CurrentUser IsNot Nothing) Then
                        Me.Notes &= "User: " & .CurrentUser.UserName & vbNewLine
                    End If
                    If (.CurrentLogin IsNot Nothing) Then
                        Me.Notes &= "LogIn: " & .CurrentLogin.RemoteIP & ": " & .CurrentLogin.RemotePort & " (" & .CurrentLogin.NomeUfficio & ")" & vbNewLine
                        Me.Notes &= "       " & Formats.FormatUserDateTime(.CurrentLogin.LogInTime) & " - " & Formats.FormatUserDateTime(.CurrentLogin.LogOutTime) & " (" & [Enum].GetName(GetType(LogOutMethods), .CurrentLogin.LogoutMethod) & ")" & vbNewLine
                        Me.Notes &= "       " & .CurrentLogin.UserAgent & vbNewLine
                    End If
                    Me.Notes &= vbNewLine
                    SyncLock .CurrentSession
                        Me.Notes &= "Cursori aperti: " & .RemoteOpenedCursors.Count & vbNewLine
                        For Each c As DBObjectCursorBase In .RemoteOpenedCursors
                            Me.Notes &= TypeName(c) & "[" & c.Token & ", " & [Enum].GetName(GetType(DBCursorStatus), c.CursorStatus) & "]: " & c.GetFullSQL & vbNewLine
                        Next
                    End SyncLock
                    Me.Notes &= vbNewLine
                    SyncLock info
                        Me.Notes &= ""
                        Me.Notes &= minidom.Sistema.Strings.NChars(80, "-") & vbNewLine
                        Me.Notes &= "QUERIS PENDENTI " & vbNewLine
                        Me.Notes &= minidom.Sistema.Strings.NChars(80, "-") & vbNewLine

                        'For Each item As StatsItem In info.QueriesInfo
                        '    Me.Notes &= Formats.FormatUserDateTime(item.LastRun) & " " & item.Name & vbNewLine
                        'Next

                        'Me.Notes &= ""
                        'Me.Notes &= minidom.Sistema.Strings.NChars(80, "-") & vbNewLine
                        'Me.Notes &= "RICHIESTE PENDENTI " & vbNewLine
                        'Me.Notes &= minidom.Sistema.Strings.NChars(80, "-") & vbNewLine

                        'For Each item As StatsItem In info.PagesInfo
                        '    Me.Notes &= Formats.FormatUserDateTime(item.LastRun) & " " & item.Name & vbNewLine
                        'Next
                    End SyncLock

                End With
            Next

        End Sub

    End Class

End Class

File: MiniDOMWebLibs/MiniDOM/WebSiteDebugInfo.cs
using System;
using DMD;

namespace minidom
{
    public partial class WebSite
    {

        /// <summary>
    /// Questa classe serve per stabilire un sistema di comunicazione ad eventi
    /// con
    /// </summary>
    /// <remarks></remarks>
        public class WebSiteDebugInfo : Sistema.SysDebuInfo
        {
            public WebSiteDebugInfo()
            {
                // DMDObject.IncreaseCounter(Me)
            }

            public override void Initialize()
            {
                base.Initialize();
                Notes += "";
                Notes += DMD.Strings.NChars(80, "-") + DMD.Strings.vbNewLine;
                Notes += "SESSIONI ATTIVE" + DMD.Strings.vbNewLine;
                Notes += DMD.Strings.NChars(80, "-") + DMD.Strings.vbNewLine;
                var col = ((WebApplicationContext)Sistema.ApplicationContext).GetAllSessions();
                foreach (CSessionInfo info in col)
                {
                    Notes += DMD.Strings.vbNewLine;
                    Notes += DMD.Strings.NChars(80, "-") + DMD.Strings.vbNewLine;
                    Notes += "SessionID: " + info.CurrentSession.SessionID + DMD.Strings.vbNewLine;
                    Notes += "StartTime: " + Sistema.Formats.FormatUserDateTime(info.CurrentSession.StartTime);
                    Notes += DMD.Strings.vbNewLine;
                    if (info.CurrentUser is object)
                    {
                        Notes += "User: " + info.CurrentUser.UserName + DMD.Strings.vbNewLine;
                    }

                    if (info.CurrentLogin is object)
                    {
                        Notes += "LogIn: " + info.CurrentLogin.RemoteIP + ": " + info.CurrentLogin.RemotePort + " (" + info.CurrentLogin.NomeUfficio + ")" + DMD.Strings.vbNewLine;
                        Notes += "       " + Sistema.Formats.FormatUserDateTime(info.CurrentLogin.LogInTime) + " - " + Sistema.Formats.FormatUserDateTime(info.CurrentLogin.LogOutTime) + " (" + Enum.GetName(typeof(Sistema.LogOutMethods), info.CurrentLogin.LogoutMethod) + ")" + DMD.Strings.vbNewLine;
                        Notes += "       " + info.CurrentLogin.UserAgent + DMD.Strings.vbNewLine;
                    }

                    Notes += DMD.Strings.vbNewLine;
                    lock (info.CurrentSession)
                    {
                        Notes += "Cursori aperti: " + info.RemoteOpenedCursors.Count + DMD.Strings.vbNewLine;
                        foreach (Databases.DBObjectCursorBase c in info.RemoteOpenedCursors)
                            Notes += DMD.RunTime.vbTypeName(c) + "[" + c.Token + ", " + Enum.GetName(typeof(Databases.DBCursorStatus), c.CursorStatus) + "]: " + c.GetFullSQL() + DMD.Strings.vbNewLine;
                    }

                    Notes += DMD.Strings.vbNewLine;
                    lock (info)
                    {
                        Notes += "";
                        Notes += DMD.Strings.NChars(80, "-") + DMD.Strings.vbNewLine;
                        Notes += "QUERIS PENDENTI " + DMD.Strings.vbNewLine;
                        Notes += DMD.Strings.NChars(80, "-") + DMD.Strings.vbNewLine;

                        // For Each item As StatsItem In info.QueriesInfo
                        // Me.Notes &= Formats.FormatUserDateTime(item.LastRun) & " " & item.Name & vbNewLine
                        // Next

                        // Me.Notes &= ""
                        // Me.Notes &= minidom.DMD.Strings.NChars(80, "-") & vbNewLine
                        // Me.Notes &= "RICHIESTE PENDENTI " & vbNewLine
                        // Me.Notes &= minidom.DMD.Strings.NChars(80, "-") & vbNewLine

                        // For Each item As StatsItem In info.PagesInfo
                        // Me.Notes &= Formats.FormatUserDateTime(item.LastRun) & " " & item.Name & vbNewLine
                        // Next
                    }
                }
            }
        }
    }
}